go/coinstacks/thorchain/api/handler.go

```diff
@@ -66,3 +145,34 @@ func (h *Handler) ParseFee(tx signing.Tx, txid string, denom string) cosmos.Valu
 
 	return fee
 }
+
+// formatTx creates a synthetic transaction from a BlockEndEvent
+func (h *Handler) formatTx(tx *ResultTx) (*cosmos.Tx, error) {
+	block, err := h.BlockService.GetBlock(int(tx.Height))
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get block: %d", tx.Height)
+	}
+
+	t := &cosmos.Tx{
+		BaseTx: api.BaseTx{
+			BlockHash:   &block.Hash,
+			BlockHeight: block.Height,
+			Timestamp:   block.Timestamp,
+		},
+		Index:         -1, // synthetic transactions don't have a real tx index
+		Fee:           cosmos.Value{Amount: "2000000", Denom: "rune"},
```

Comment:
TODO: need to actually get fees somehow...

Reply:
https://github.com/shapeshift/unchained/pull/512/commits/e4b5c8f287d57c4a56385a41e602f7ea748831cb